pkg/apis/internalversion/metric_type.go

```diff
@@ -43,16 +43,42 @@ type MetricConfig struct {
 	Name string
 	// Help provides information about this metric.
 	Help string
-	// Kind is kind of metric (ex. counter, gauge, histogram).
-	Kind string
+	// Kind is kind of metric.
+	Kind Kind
 	// Labels are metric labels.
 	Labels []MetricLabel
 	// Value is a CEL expression.
 	Value string
 	// Buckets is a list of buckets for a histogram metric.
 	Buckets []MetricBucket
+	// Dimension is a dimension of the metric.
+	Dimension Dimension
```

Comment:
I want to simulate some custom metrics, such as (jobs.batch/httpserver_requests_qps), how should I do it?

Reply:
https://github.com/kubernetes-sigs/kwok/pull/669/files#diff-30e09ea9066ffe73bdf2a74764fe57f89955f2a68efa9dff0380cf5094e2143aR52-R129 
Using the configuration to describe indicators

Reply:
```go 
// Dimension is a dimension of the metric.
// +enum
type Dimension string

const (
	// DimensionNode is a node dimension.
	DimensionNode Dimension = "node"
	// DimensionPod is a pod dimension.
	DimensionPod Dimension = "pod"
	// DimensionContainer is a container dimension.
	DimensionContainer Dimension = "container"
)
```

My point is that my custom indicator may not be any of these three dimensions, so how should I simulate it?

Reply:
This PR is just for support `/metrics/resource` https://github.com/kubernetes-sigs/kwok/issues/671

Other indicators in the planning https://github.com/kubernetes-sigs/kwok/issues/670

Reply:
Ok, thinks for reminder.